dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocityRequestWrapper.java

```diff
@@ -1,347 +1,149 @@
-		return _request.getContentLength();
-	}
-
-	public String getContentType() {
-		return _request.getContentType();
-	}
-
-	public ServletInputStream getInputStream() throws IOException {
-		return _request.getInputStream();
-	}
-
-	public String getLocalAddr() {
-		return _request.getLocalAddr();
-	}
-
-	public String getLocalName() {
-		return _request.getLocalName();
-	}
-
-	public int getLocalPort() {
-		return _request.getLocalPort();
-	}
-
-	public Locale getLocale() {
-		return _request.getLocale();
-	}
 
-	public Enumeration getLocales() {
-		return _request.getLocales();
-	}
-
-	public String getParameter(String arg0) {
-		String ret = _request.getParameter(arg0);
-		if(UtilMethods.isSet(ret) && Xss.URLHasXSS(ret)){
-			ret = UtilMethods.htmlifyString(ret);
-		}
-		return ret;
-	}
-
-	public Map getParameterMap() {
-		return _request.getParameterMap();
-	}
-
-	public Enumeration getParameterNames() {
-		return _request.getParameterNames();
-	}
-
-	public String[] getParameterValues(String arg0) {
-		return _request.getParameterValues(arg0);
-	}
-
-	public String getProtocol() {
-		return _request.getProtocol();
-	}
-
-	public BufferedReader getReader() throws IOException {
-		return _request.getReader();
-	}
-
-	public String getRealPath(String arg0) {
-		return arg0;
-	}
-
-	public String getRemoteAddr() {
-		return _request.getRemoteAddr();
-	}
-
-	public String getRemoteHost() {
-		return _request.getRemoteHost();
-	}
-
-	public int getRemotePort() {
-		return _request.getRemotePort();
-	}
-
-	public RequestDispatcher getRequestDispatcher(String uri) {
-		return _request.getRequestDispatcher(uri);
-	}
-
-	public String getScheme() {
-		return _request.getScheme();
-	}
-
-	public String getServerName() {
-		return _request.getServerName();
-	}
-
-	public int getServerPort() {
-		return _request.getServerPort();
-	}
-
-	public boolean isSecure() {
-		return _request.isSecure();
-	}
-
-	public void removeAttribute(String arg0) {
-	    // do nothing
-	}
-
-	public void setAttribute(String arg0, Object arg1) {
-	    if(!SET_VALUE_BLACKLIST.contains(arg0)) {
-	        _request.setAttribute(arg0, arg1);
-	    }
-	}
-
-	public void setCharacterEncoding(String arg0) throws UnsupportedEncodingException {
-		_request.setCharacterEncoding(arg0);
-	}
-
-    public AsyncContext getAsyncContext() {
-        return _request.getAsyncContext();
-    }
-
-    public DispatcherType getDispatcherType() {
-        return _request.getDispatcherType();
+    @Override
+    public String getRequestURI() {
+        if (this.dotCMSUri != null) {
+            return this.dotCMSUri;
+        }
+        return super.getRequestURI();
     }
 
-    public ServletContext getServletContext() {
-        SecurityLogger.logInfo(this.getClass(), "User trying to access ServletContext from Velocity");
-        return null;
+    @Override
+    public StringBuffer getRequestURL() {
+        return new StringBuffer(super.getScheme() + "://" + super.getServerName() + ":"
```

Comment:
Codacy found an issue: [Avoid concatenating nonliterals in a StringBuffer/StringBuilder constructor or append().](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5831320)